tests/e2e/declarative_database_management_test.go

```diff
@@ -154,13 +159,83 @@ var _ = Describe("Declarative databases management test", Label(tests.LabelSmoke
 					Expect(utils.DeleteObject(env, database)).To(Succeed())
 				})
 
-				By("verifying database is still existing", func() {
+				By("verifying db is still existing", func() {
 					primaryPodInfo, err := env.GetClusterPrimary(namespace, clusterName)
 					Expect(err).ToNot(HaveOccurred())
 
 					assertDatabaseExists(namespace, primaryPodInfo.Name, dbname, true)
 				})
 			})
 		})
+		When("Database CRD reclaim policy is set to delete inside spec", func() {
+			It("can add a declarative database", func() {
+				By("applying Database CRD manifest", func() {
+					CreateResourceFromFile(namespace, databaseManifestWithDeleteReclaimPolicy)
+					databaseObjectName, err = env.GetResourceNameFromYAML(databaseManifestWithDeleteReclaimPolicy)
+					Expect(err).NotTo(HaveOccurred())
+				})
+				By("ensuring the Database CRD succeeded reconciliation", func() {
+					// get database object
+					databaseWithDeleteRetainPolicy = &apiv1.Database{}
+					databaseNamespacedName := types.NamespacedName{
+						Namespace: namespace,
+						Name:      databaseObjectName,
+					}
+
+					Eventually(func(g Gomega) {
+						err := env.Client.Get(env.Ctx, databaseNamespacedName, databaseWithDeleteRetainPolicy)
+						Expect(err).ToNot(HaveOccurred())
+						g.Expect(databaseWithDeleteRetainPolicy.Status.Ready).Should(BeTrue())
+					}, 300).WithPolling(10 * time.Second).Should(Succeed())
+				})
+
+				By("verifying the db exists", func() {
+					primaryPodInfo, err := env.GetClusterPrimary(namespace, clusterName)
+					Expect(err).ToNot(HaveOccurred())
+
+					assertDatabaseExists(namespace, primaryPodInfo.Name, dbname, true)
+				})
+			})
+
+			It("maintains the db when Database CRD is removed", func() {
```

Comment:
wait ... this is exactly the opposite. Inside it's checking the Postgres DB does NOT exist, line 209

Reply:
You are right, it should be "does not maintain..."